modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-fragment.glsl.js

```diff
@@ -36,23 +38,37 @@ varying vec2 uv;
 void main(void) {
   geometry.uv = uv;
 
-  vec4 texColor = texture2D(iconsTexture, vTextureCoords);
-  
-  float alpha = texColor.a;
+  float alpha = texture2D(iconsTexture, vTextureCoords).a;
+
   // if enable sdf (signed distance fields)
   if (sdf) {
-    float distance = texture2D(iconsTexture, vTextureCoords).a;
-    alpha = smoothstep(buffer - vGamma, buffer + vGamma, distance);
+    alpha = smoothstep(buffer - vGamma, buffer + vGamma, alpha);
   }
 
   // Take the global opacity and the alpha from vColor into account for the alpha component
   float a = alpha * vColor.a;
-
+  
   if (a < alphaCutoff) {
-    discard;
+    // We are now in the background, let's decide what to draw
+    if (shouldDrawBackground && !picking_uActive) {
+      // draw background color and return if not picking
+      gl_FragColor = vec4(backgroundColor.rgb, vColor.a);
+      return;
+    } else if (picking_uActive) {
+      // allow picking to work and pick the background
+      DECKGL_FILTER_COLOR(gl_FragColor, geometry);
```

Comment:
Just let this fall through - the perf impact is minimal.

Reply:
ok...